subprojects/model-core/src/main/java/org/gradle/internal/instantiation/AsmBackedClassGenerator.java

```diff
@@ -510,6 +543,12 @@ private void initializeFields(MethodVisitor methodVisitor) {
                 methodVisitor.visitMethodInsn(INVOKESTATIC, ASM_BACKED_CLASS_GENERATOR_TYPE.getInternalName(), "getServicesForNext", RETURN_SERVICE_LOOKUP, false);
                 methodVisitor.visitFieldInsn(PUTFIELD, generatedType.getInternalName(), SERVICES_FIELD, SERVICE_LOOKUP_TYPE.getDescriptor());
             }
+            if (requiresInstantiator) {
+                // this.instantiator = AsmBackedClassGenerator.getInstantiatorForNext()
```

Comment:
this is still WIP, right?

Reply:
Yes it is. The PR was created before draft PRs existed ;)
The comment is correct, though. It is what is done in the bytecode.